app/controlplane/plugins/core/dependency-track/v1/extension_test.go

```diff
@@ -335,3 +338,88 @@ func TestValidateAttachmentConfiguration(t *testing.T) {
 		}
 	}
 }
+
+func TestVerifyAllFilters(t *testing.T) {
+	attestationAnnotations := map[string]string{
+		"environment": "prod",
+		"team":        "security",
+	}
+
+	materialAnnotations := map[string]string{
+		"environment": "staging",
+		"critical":    "true",
+	}
+
+	testCases := []struct {
+		name    string
+		filter  string
+		wantErr bool
```

Comment:
nitpick. I'd not call it wantErr, just `want`, otherwise it can be interpreted as you want to return an error in golang :)

Nice tests btw :)